lib/Haskell/Data/List.agda

```diff
@@ -0,0 +1,103 @@
+module Haskell.Data.List where
+
+open import Haskell.Prelude
+
+open import Haskell.Data.Ord using (comparing)
+
+open import Haskell.Law.Eq
+open import Haskell.Law.Equality
+
+{-----------------------------------------------------------------------------
+    Operations
+------------------------------------------------------------------------------}
+
+partition : (a → Bool) → List a → (List a × List a)
+partition p xs = (filter p xs , filter (not ∘ p) xs)
+
+deleteBy : (a → a → Bool) → a → List a → List a
+deleteBy eq x ys = filter (not ∘ (eq x)) ys
+
+delete : ⦃ Eq a ⦄ → a → List a → List a
+delete = deleteBy (_==_)
```

Comment:
As mentioned by @HeinrichApfelmus in private correspondence, this implementation deletes *all* elements instead of the *first* one as the Haskell prelude one does.

So, definitely rename this to `deleteAll` and think of adding the correct `delete` as well.

Reply:
I have chosen to rename this to `deleteAll` and also specialize it to `IsLawfulEq`.

I have also decided to not add `delete` at this time if that's ok with you.